src/errors/memory.ts

```diff
@@ -1,9 +1,21 @@
 export class MemoryError extends Error {}
 
-export const UnknownAddressError =
-  'Access memory at unknown or uninitialized address';
-export const WriteOnceError =
-  'Write existing memory. Can only write to memory once.';
-export const SegmentIncrementError = 'Error incrementing number of segments';
-export const InstructionError = 'Instruction must be a Field Element';
-export const DeductionError = 'Failed to deduce operands';
+/** Write existing memory. Can only write to memory once */
+export class WriteOnceError extends MemoryError {}
```

Comment:
See the Cairo paper

> It turns out that, for most of the memory accesses in programs, a Nondeterministic Read-Only Memory is sufficient (for example, Section 6 explains
how a function stack may be implemented in the read-only model), and in
places where it does not suffice, it’s possible to simulate a full read-write
memory using it (see Section 8.5). Therefore, Cairo uses a nondeterministic
read-only memory as its memory model.

Reply:
to be more accurate, the prover can write once, but the cairo vm is read-only if this makes sense. So the error may stay `WriteOnce` depending on the context

Reply:
Indeed, `WriteOnce` doesn't fit perfectly

Renaming it `InconsistentMemoryError` as in the Rust/Python VM seems better as it conveys the read-only status of the memory in the cairo vm:
- Once a cell value is constrained (i.e. not undefined), attempting to make read another value from this cell throws `InconsistentMemoryError`

But I would also rename the [`write()`](https://github.com/kkrt-labs/cairo-vm-ts/blob/49159f2cc352c06fda4e4c5cb306ff3228150961/src/memory/memory.ts#L33) function then

Is `read()` more meaningful? (python vm defines `__setitem__` while rust vm declares `insert()` which is equivalent to `write()` imo